channeldb/kvdb/etcd/bucket.go

```diff
@@ -0,0 +1,29 @@
+package etcd
+
+import (
+	"bytes"
+)
+
+func isValidKey(key []byte) bool {
+	return !bytes.ContainsAny(key, "/~")
+}
+
+func makeBucketPrefix(path string) string {
+	return "b/" + path + "~"
```

Comment:
Any reason the tilde is chosen? For the purpose of quickly terminating certain classes of range queries?

Reply:
This (tilde may be anything really) is just to simplify the case where we iterate trough buckets in the current bucket. Instead of making etcd fetch all bucket keys with the path prefix, we just select the ones in the next level.